### ISSUE TITLE
different behavior with mapping of  `either` and mapping of `map` when mapping is a closure

### ISSUE THREAD

**Issue by k33g** [eclipse/golo-lang#358]
When I pass a closure to an `Option`, I get different behavior with mapping of  `either` and mapping of `map`. Perhaps i miss something?

``` golo
try {
  raise("Huston?")
} catch(exception) {
  Option(|ex|-> println(ex)): either(|cbk| {
    cbk(exception)
  },{
    # nothing
  })
}


try {
  raise("Huston?")
} catch(exception) {
  # here I get a "java.lang.IllegalArgumentException: not a direct method handle"
  Option(|ex|-> println(ex)): map(|cbk| {
    cbk(exception)
  })
}
```

**Comment by yloiseau** [comment]
Known bug. `map` is a native java method, while `either` is an augmentation. The issue raise since you pass a "real closure" (i.e. with a closed value) to a native method expecting a java `Function` instance. The implicit conversion fails. See #277

**Comment by yloiseau** [comment]
As a side note, what you need here is an applicative functor... see the `apply` augmentation on `Optional` :smile:

**Comment by yloiseau** [comment]
(closed as duplicate of #277)
